src/robotFactory.js

```diff
@@ -1,10 +1,79 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
 
-class FlyingRobot {}
+    if (!this.coords.x) {
+      this.coords.x = 0;
+    }
 
-class DeliveryDrone {}
+    if (!this.coords.y) {
+      this.coords.y = 0;
+    }
+
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return 'Robot: ' + this.name + ',' + ' Chip version: '
+    + this.chipVersion + ',' + ' Weight: ' + this.weight;
+  }
```

Comment:
```suggestion
  getInfo() {
    return `Robot: ${this.name}, `
      + `Chip version: ${this.chipVersion}, `
      + `Weight: ${this.weight}`;
  }
```
Можна було б скомбінувати інтерполяцію і конкатинацію